jms/src/main/scala/akka/stream/alpakka/jms/JmsConnector.scala

```diff
@@ -30,10 +35,6 @@ private[jms] trait JmsConnector[S <: JmsSession] { this: GraphStageLogic =>
 
   protected val fail: AsyncCallback[Throwable] = getAsyncCallback[Throwable](e => failStage(e))
 
-  private val onConnection: AsyncCallback[jms.Connection] = getAsyncCallback[jms.Connection] { c =>
-    jmsConnection = Some(c)
-  }
```

Comment:
Removed the connection setting using `AsyncCallback` and set directly instead. Some tests rely on the connection to be immediately available after establishing and fail if not. We minimize the risk of such failure. Also connection is only accessed at startup and shutdown. Making it `@volatile` does not incur runtime overhead.

Reply:
Hmm, apparently this is still not enough. That test is still failing despite tightening up the timeline.

Reply:
OK. Need to guarantee the connection is closed in an async system. To do so, we need to hold the connection in a future to ensure the connection will be closed even if it is resolved later than the `postStop` is called. This should settle it. But this test has creeped the scope of change.